magda/config_reader.py

```diff
@@ -146,13 +153,25 @@ def _validate_config_parameters_structure(config_parameters):
                     f"string or numeric type. Found value: {value}."
                 )
 
+    @staticmethod
+    def _check_pipeline_name(name: str):
+        if not isinstance(name, str):
+            raise WrongParameterValueException(
+                "Parameter 'name' in config should accept string values only. "
+                f"Found value: '{name}'."
+            )
+
     @staticmethod
     def _extract_information_from_yaml(parsed_yaml, shared_parameters):
         try:
             modules = [ConfigReader.ConfigModule(**data) for data in parsed_yaml['modules']]
         except TypeError:
             raise Exception("Every module defined in a config file has to have a name and a type.")
 
+        pipeline_name = None
+        if 'name' in parsed_yaml:
+            pipeline_name = parsed_yaml['name']
```

Comment:
```suggestion
        pipeline_name = parsed_yaml.get('name')
```
Assuming that `parsed_yaml` is dict (https://stackoverflow.com/a/11041421)